src/Magento/FunctionalTestingFramework/Util/ModulePathExtractor.php

```diff
@@ -11,41 +11,92 @@
  */
 class ModulePathExtractor
 {
-    const MAGENTO = 'Magento';
+    const SPLIT_DELIMITER = '_';
+
+    /**
+     * Test module paths
+     *
+     * @var array
+     */
+    private $testModulePaths = [];
+
+    /**
+     * ModulePathExtractor constructor
+     */
+    public function __construct()
+    {
+        $verbose = true;
+        if (empty($this->testModulePaths)) {
```

Comment:
Remove this conditional, `$this->testModulePaths` is never not going to be empty.

Reply:
This is intentional. $this->testModulePaths will be queried one time and is cached after

Reply:
I don't think so; every place that ModulePathExtractor is used, it's instantiated individually. Since this class isn't a singleton, the instance of the class will always have a nulled out `$this->testModulePaths`.

You can see this by attaching a debugger to line 30; that line is hit multiple times in a single `generate:tests` command.